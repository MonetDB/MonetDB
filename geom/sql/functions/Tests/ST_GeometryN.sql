select ST_GeometryN(ST_mpointfromtext('multipoint(1 2 7, 3 4 7, 5 6 7, 8 9 10)'), 4);
select ST_GeometryN(ST_mpointfromtext('multipoint(1 2 7, 3 4 7, 5 6 7, 8 9 10)'), 0);
select ST_GeometryN(ST_mpointfromtext('multipoint(1 2 7, 3 4 7, 5 6 7, 8 9 10)'), 1);

SELECT ST_GeometryN(
 ST_GeomCollFromText(
  'geometryCollection(
    polygon((2.5 2.5,4.5 2.5, 3.5 3.5, 2.5 2.5), (10 11, 12 11, 11 12, 10 11)), 
    multipoint(10 10, 20 20, 30 30))'), 
  4
);

SELECT ST_GeometryN(
 ST_GeomCollFromText(
  'geometryCollection(
    polygon((2.5 2.5,4.5 2.5, 3.5 3.5, 2.5 2.5), (10 11, 12 11, 11 12, 10 11)), 
    multipoint(10 10, 20 20, 30 30))'),
  2
);

SELECT ST_GeometryN(geom, 2) AS "GEOMETRY 2" FROM geometries WHERE id IN (19,20,21,22,23,24,25);

 

