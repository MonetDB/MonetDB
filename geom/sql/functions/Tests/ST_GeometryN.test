query T rowsort
select ST_GeometryN(ST_mpointfromtext('multipoint(1 2 7, 3 4 7, 5 6 7, 8 9 10)'), 4)
----
POINT Z (8 9 10)

query T rowsort
select ST_GeometryN(ST_mpointfromtext('multipoint(1 2 7, 3 4 7, 5 6 7, 8 9 10)'), 0)
----
NULL

query T rowsort
select ST_GeometryN(ST_mpointfromtext('multipoint(1 2 7, 3 4 7, 5 6 7, 8 9 10)'), 1)
----
POINT Z (1 2 7)

query T rowsort
SELECT ST_GeometryN(
 ST_GeomCollFromText(
  'geometryCollection(
    polygon((2.5 2.5,4.5 2.5, 3.5 3.5, 2.5 2.5), (10 11, 12 11, 11 12, 10 11)),
    multipoint(10 10, 20 20, 30 30))'),
  4
)
----
NULL

query T rowsort
SELECT ST_GeometryN(
 ST_GeomCollFromText(
  'geometryCollection(
    polygon((2.5 2.5,4.5 2.5, 3.5 3.5, 2.5 2.5), (10 11, 12 11, 11 12, 10 11)),
    multipoint(10 10, 20 20, 30 30))'),
  2
)
----
MULTIPOINT (10 10, 20 20, 30 30)

query T rowsort
SELECT ST_GeometryN(geom, 2) AS "GEOMETRY 2" FROM geometries WHERE id IN (19,20,21,22,23,24,25)
----
LINESTRING Z (10 20 10, 30 40 20)
LINESTRING Z (10 20 10, 30 40 20, 10 20 10)
LINESTRING Z (50 60 30, 60 70 40)
LINESTRING Z (50 60 40, 60 70 50)
LINESTRING Z (50 60 70, 40 50 60, 20 30 50, 50 60 40)
NULL
POINT Z (30 40 30)

