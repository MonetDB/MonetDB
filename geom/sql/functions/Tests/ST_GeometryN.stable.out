stdout of test 'ST_GeometryN` in directory 'geom/sql/functions` itself:


# 15:06:10 >  
# 15:06:10 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=32276" "--set" "mapi_usock=/var/tmp/mtest-10872/.s.monetdb.32276" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/export/scratch1/alvanaki/INSTALL/MonetDB-public/var/MonetDB/mTests_geom_sql_Tests_functions" "--set" "mal_listing=0"
# 15:06:10 >  

# MonetDB 5 server v11.20.0
# This is an unreleased version
# Serving database 'mTests_geom_sql_Tests_functions', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 15.356 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2014 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://sibuyan.da.cwi.nl:32276/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-10872/.s.monetdb.32276
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded


# 15:06:10 >  
# 15:06:10 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-10872" "--port=32276"
# 15:06:10 >  

#select ST_GeometryN(ST_mpointfromtext('multipoint(1 2 7, 3 4 7, 5 6 7, 8 9 10)'), 4);
% . # table_name
% %1 # name
% geometry # type
% 0 # length
[ "POINT Z (8 9 10)"	]
#select ST_GeometryN(ST_mpointfromtext('multipoint(1 2 7, 3 4 7, 5 6 7, 8 9 10)'), 0);
% . # table_name
% %1 # name
% geometry # type
% 0 # length
[ NULL	]
#select ST_GeometryN(ST_mpointfromtext('multipoint(1 2 7, 3 4 7, 5 6 7, 8 9 10)'), 1);
% . # table_name
% %1 # name
% geometry # type
% 0 # length
[ "POINT Z (1 2 7)"	]
#SELECT ST_GeometryN(
# ST_GeomCollFromText(
#  'geometryCollection(
#    polygon((2.5 2.5,4.5 2.5, 3.5 3.5, 2.5 2.5), (10 11, 12 11, 11 12, 10 11)), 
#    multipoint(10 10, 20 20, 30 30))'), 
#  4
#);
% . # table_name
% %1 # name
% geometry # type
% 0 # length
[ NULL	]
#SELECT ST_GeometryN(
# ST_GeomCollFromText(
#  'geometryCollection(
#    polygon((2.5 2.5,4.5 2.5, 3.5 3.5, 2.5 2.5), (10 11, 12 11, 11 12, 10 11)), 
#    multipoint(10 10, 20 20, 30 30))'),
#  2
#);
% . # table_name
% %1 # name
% geometry # type
% 0 # length
[ "MULTIPOINT (10 10, 20 20, 30 30)"	]
#SELECT ST_GeometryN(geom, 2) AS "GEOMETRY 2" FROM geometries WHERE id IN (19,20,21,22,23,24,25);
% sys. # table_name
% "GEOMETRY 2" # name
% geometry # type
% 0 # length
[ "POINT Z (30 40 30)"	]
[ "LINESTRING Z (50 60 30, 60 70 40)"	]
[ "LINESTRING Z (50 60 40, 60 70 50)"	]
[ "LINESTRING Z (50 60 70, 40 50 60, 20 30 50, 50 60 40)"	]
[ NULL	]
[ "LINESTRING Z (10 20 10, 30 40 20)"	]
[ "LINESTRING Z (10 20 10, 30 40 20, 10 20 10)"	]

# 15:06:11 >  
# 15:06:11 >  "Done."
# 15:06:11 >  

